### DISCUSSION TITLE
Reuse  host's SSH agent if `SSH_AUTH_SOCK` is set

### DISCUSSION THREAD

**Discussion by ZauberNerd** [actions/checkout#660] [Ideas]
I'm wondering whether a PR for this feature would be accepted?

I understand that this is quite an edge-case, as the regular GitHub runners, do not have an SSH agent running and since the `actions/checkout` action accepts an SSH private key, it wouldn't be of much use to most users.

In my case I want to be able to use [`act`](https://github.com/nektos/act) without having to supply a personal access token and I have my SSH key on a hardware key, so even if I wanted to, I couldn't supply it to the checkout action.

Another use-case I can think of might be custom GitHub runners, which could set-up an `ssh-agent` beforehand, although I'm not using custom runners and don't know off the top of my head, if that would make sense.

So basically the feature would work as follows:
- check if `SSH_AUTH_SOCK` is set
- rewrite git url to `ssh://git@github.com/owner/repo.git`
- (and maybe some more steps, since I've seen that there is a lot of code around authentication already: https://github.com/actions/checkout/blob/main/src/git-auth-helper.ts)

In case such a PR would be accepted, I could look into implementing it, but I wanted to get feedback first.

**Comment by ZauberNerd** [comment]
Ping @ericsciple @ethomson @TingluoHuang as you seem to be fairly active / members of the actions org. Any thoughts on this?
